src/app/Library/CrudPanel/Traits/Read.php

```diff
@@ -203,17 +203,24 @@ public function getDefaultPageLength()
      */
     public function addCustomPageLengthToPageLengthMenu()
     {
-        // If the default Page Length isn't in the menu's values, Add it the beginnin and resort all to show a croissant list.
-        // assume both arrays are the same length.
-        if (! in_array($this->getDefaultPageLength(), $this->getOperationSetting('pageLengthMenu')[0])) {
-            // Loop through 2 arrays of prop. pageLengthMenu
-            foreach ($this->getOperationSetting('pageLengthMenu') as $key => &$page_length_choices) {
-                // This is a condition that should be always true.
-                if (is_array($page_length_choices)) {
-                    array_unshift($page_length_choices, $this->getDefaultPageLength());
-                }
+        $values = $this->getOperationSetting('pageLengthMenu')[0];
+        $labels = $this->getOperationSetting('pageLengthMenu')[1];
+
+        // This is a condition that should be always true.
+        if (is_array($values) && is_array($labels)) {
+            $position = array_search($this->getDefaultPageLength(), $values);
+            if ($position !== false) {
+                $defaultLabel = isset($labels[$position]) ? $labels[$position] : $this->getDefaultPageLength();
+                array_unshift($values, $this->getDefaultPageLength());
+                array_unshift($labels, $defaultLabel);
+
+                //now make it unique.
+                $values = array_unique($values);
+                $labels = array_unique($labels);
             }
         }
+
+        $this->setOperationSetting('pageLengthMenu', [$values, $labels]);
```

Comment:
@pxpm do we need this function?

Because all the logic is inside `setPageLengthMenu` and with this function code is not working 😅

When I reverted this, it worked once again.

Reply:
Setting up with 0 will fail in the database in the stock backpack search functionality, but if you override, developer could use the 0 as a placeholder for something and handle it on his side. 

Like we talked before, I don't think that a regular dev that wants to setup a paginator will use 0, but I think we should not error on '0' because developer could handle it on his side. 

On the other hand, if this 0 handling starts to get out of control I think we better add to the docs that `0` is not supported and remove everything that is related to handle `0` as a key in the array. 
**I am starting feeling that this is the way to go instead of adding that much code to handle `0`**

What do you think @promatik ?

Reply:
About the function itself, that function exists because you can simply `add a new value to the backpack paginator and make it default` without the need to go copy the whole backpack paginatior and add the value manuay. 

You can just config `CRUD::setOperationSetting('defaultPageLength', 30)` in your ListOperation and we will pick it and add it to paginator making it the default.

Reply:
@pxpm I agree that we should avoid to add this extra layer os code/conditions to handle "0", but what should developers use for the "All"?

Regarding the function, now I know 😅, but something is not working 😬
I may later have a look once again 👍

Reply:
Hello @promatik, to emulated the `all` we used `-1` before, that's why I also added tests for -1. 

I am going to have a talk with @tabacitu about this, I think we should remove the extra coded I added to support 0, and just let devs known that `0` is a non-valid paginator number, and to use -1 if they want to show all.

I will come back to you asap 👍 thanks again @promatik

Reply:
Like we discussed, let's not support 0, there is no sense in supporting it.